tests/interfaces/default_qubit_2_integration/test_autograd_qnode_default_qubit_2.py

```diff
@@ -94,9 +94,9 @@ def circuit(a):
 
         assert circuit.qtape.interface is None
 
-        # without the interface, the QNode simply returns a scalar array
-        assert isinstance(res, np.ndarray)
-        assert res.shape == tuple()  # pylint: disable=comparison-with-callable
+        # without the interface, the QNode simply returns a scalar array or float
+        assert isinstance(res, (np.ndarray, float))
+        assert qml.math.shape(res) == tuple()  # pylint: disable=comparison-with-callable
```

Comment:
This is not optimal, I suppose? Shouldn't the QNode return value be independent of the device? :thinking: Sorry if this is an established behaviour, I may just not be used to it.

Reply:
Good question. @albi3ro and I have discussed this briefly offline, and we decided it was okay to have `float` as a valid dtype for scalar returns with no interface.

Reply:
Fair enough! I hope this does not break any further processing, but as lightning got its own VJP method, I suppose it is fine :) Is this documented anywhere? :)

Reply:
Not currently. I will discuss with core team offline and update docs if we decide to.

Reply:
@albi3ro  has a doc PR that will address this.